contents/security/sqlInjection.md

```diff
@@ -13,45 +13,87 @@
+```
+will only returns the records in table "users" whose user name is "Admin". Here the parameter `Username = Admin` in the `WHERE clause` works as a filter.
+
+More information about SQL can be found [here](https://www.w3schools.com/sql/).
+
+### SQL Injection
+SQL injection is one of the most common web hacking techniques. It replaces malicious code in SQL statements via web page input. Websites combine SQL syntax with user input to query the underlying database of the website. However, some websites may not check the syntax of user input rigorously, and therefore a malicious user can inject SQL query via the user input.
+
+## Why is it Important to Prevent SQL Injection？
+
+1. **SQL injection is frequently exhibited.** According to [Open Web Application Security Project (OWASP) report](https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf), the injection attack is always the annual top 1 application security risk from 2013 to 2017. In addition, [Statistics from Akamai](https://www.akamai.com/uk/en/resources/our-thinking/state-of-the-internet-report/web-attack-visualization.jsp) shows that in one week, over 80% of attacks are SQL injection.
+1. **SQL injection can have serious consequences.** SQL injection can cause the loss of large amount of money. Based on the [Global Threat Intelligence Report](https://www.helpnetsecurity.com/2014/03/28/analysis-of-three-billion-attacks-reveals-sql-injections-cost-196000/), even a small SQL injection attack may cause hundreds of thousands dollars lost. 
+Furthermore, information disclosure is another serious consequence. [An SQL injection attack on the toymaker company, VTech](https://coar.risc.anl.gov/consequences-of-sql-injection-attacks/), caused millions of parents' and children's profiles to be stolen. 
+Thus, we need to pay attention to prevent this attack in our code.
 
-Websites combine SQL syntax with user input to query the underlying database of the website. For example, blog websites may ask for a username and password from users for them to log in to the site. This input is used to create the SQL statement that will be sent to the database. A website which is vulnerable to SQL injection attacks may construct the SQL query like this:
+## How does SQL Injection Work?
+
+### CIA: Confidentiality, Integrity, Availability
+Before looking through SQL injection, we first look at three main requirements of computer security: confidentiality, integrity, and availability. We can understand them intuitively as follow:
+* **Confidentiality**: unauthorized people cannot view the information.
+* **Integrity**: unauthorized people cannot modify the information.
+* **Availability**: unauthorized people cannot prevent authorized people accessing the information.
+
+It turns out that SQL injection can compromise all these three security requirements. 
+We will illustrate this through two scenarios: inject more [parameters](#knowledge) to a query and inject more [queries](#knowledge).
```

Comment:
Is this section necessary? It could be seen as the author trying to showcase his/her knowledge rather than help the reader understand the topic.